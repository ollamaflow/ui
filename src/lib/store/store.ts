import { configureStore } from '@reduxjs/toolkit';
import rootReducer, { apiMiddleWares } from './rootReducer';

export const makeStore = () => {
  return configureStore({
    reducer: rootReducer,
    middleware: (gDM: any) =>
      gDM({
        serializableCheck: false,
      }).concat(apiMiddleWares),
  });
};

export type AppStore = ReturnType<typeof makeStore>;

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof rootReducer>;
export type AppDispatch = AppStore['dispatch'];
