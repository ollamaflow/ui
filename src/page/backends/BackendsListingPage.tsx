/* eslint-disable react-hooks/exhaustive-deps */
"use client";
import React, { useState, useEffect } from "react";

import { ReloadOutlined, PlusOutlined } from "@ant-design/icons";
import PageContainer from "#/components/base/pageContainer/PageContainer";
import OllamaFlowTable from "#/components/base/table/Table";
import FallBack from "#/components/fallback/FallBack";
import {
  useGetBackendQuery,
  useDeleteBackendMutation,
} from "#/lib/store/slice/apiSlice";
import { columns } from "./constants";
import OllamaFlowFlex from "#/components/base/flex/Flex";
import OllamaFlowText from "#/components/base/typograpghy/Text";
import OllamaFlowButton from "#/components/base/button/Button";
import { TableColumnType, Modal, message } from "antd";
import { Backend } from "#/lib/store/slice/types";
import { useRouter } from "next/navigation";
import { paths } from "#/constants/constant";

// Extend Window interface to include our custom handlers
declare global {
  interface Window {
    deleteBackendHandler?: (backend: Backend) => void;
  }
}

const BackendsListingPage: React.FC = () => {
  const router = useRouter();
  const [isDeleteModalVisible, setIsDeleteModalVisible] = useState(false);
  const [deletingBackend, setDeletingBackend] = useState<Backend | null>(null);

  const {
    data: backends = [],
    isLoading,
    isFetching,
    isError,
    refetch,
  } = useGetBackendQuery(null);
  const isBackendLoading = isLoading || isFetching;

  const [deleteBackend, { isLoading: isDeleteLoading }] =
    useDeleteBackendMutation();

  // Handle create backend navigation
  const handleCreateBackend = () => {
    router.push(paths.DashboardCreateBackend);
  };

  // Handle delete backend
  const handleDeleteBackend = (backend: Backend) => {
    setDeletingBackend(backend);
    setIsDeleteModalVisible(true);
  };

  // Set up global handler for delete button clicks
  useEffect(() => {
    window.deleteBackendHandler = handleDeleteBackend;
    return () => {
      delete window.deleteBackendHandler;
    };
  }, [handleDeleteBackend]);

  // Handle delete confirmation
  const handleDeleteConfirm = async () => {
    if (!deletingBackend) return;

    try {
      await deleteBackend(deletingBackend.Identifier).unwrap();
      message.success("Backend deleted successfully");
      setIsDeleteModalVisible(false);
      setDeletingBackend(null);
      // Refresh the backends list
      refetch();
    } catch (error) {
      console.error("Failed to delete backend:", error);
      message.error("Failed to delete backend");
    }
  };

  // Handle modal close
  const handleDeleteModalClose = () => {
    setIsDeleteModalVisible(false);
    setDeletingBackend(null);
  };

  if (isError) {
    return <FallBack retry={refetch} />;
  }

  return (
    <PageContainer
      pageTitle={
        <OllamaFlowFlex align="center" gap={10}>
          <OllamaFlowText>Backends</OllamaFlowText>
          <ReloadOutlined onClick={refetch} title="Refresh" />
        </OllamaFlowFlex>
      }
      pageTitleRightContent={
        <OllamaFlowButton
          type="primary"
          variant="link"
          icon={<PlusOutlined />}
          onClick={handleCreateBackend}
        >
          Create Backend
        </OllamaFlowButton>
      }
    >
      <OllamaFlowTable
        columns={columns as TableColumnType<Partial<Backend>>[]}
        dataSource={backends}
        loading={isBackendLoading}
        rowKey="Identifier"
        scroll={{ x: 1600 }}
        pagination={{
          showSizeChanger: true,
          showQuickJumper: true,
          showTotal: (total, range) =>
            `${range[0]}-${range[1]} of ${total} backends`,
        }}
        size="middle"
      />

      {/* Delete Confirmation Modal */}
      <Modal
        title="Delete Backend"
        open={isDeleteModalVisible}
        onOk={handleDeleteConfirm}
        onCancel={handleDeleteModalClose}
        confirmLoading={isDeleteLoading}
        okText="Delete"
        cancelText="Cancel"
        okButtonProps={{ danger: true }}
      >
        {deletingBackend && (
          <p>
            Are you sure you want to delete the backend{" "}
            <strong>&ldquo;{deletingBackend.Name}&rdquo;</strong> (
            {deletingBackend.Identifier})?
          </p>
        )}
      </Modal>
    </PageContainer>
  );
};

export default BackendsListingPage;
