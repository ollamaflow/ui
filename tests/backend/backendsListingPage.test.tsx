import "@testing-library/jest-dom";
import React from "react";
import { screen, waitFor } from "@testing-library/react";

import { setupServer } from "msw/node";
import { commonHandlers } from "../handler";
import BackendsListingPage from "#/page/backends/BackendsListingPage";
import { renderWithRedux } from "../store/utils";
import { createMockInitialState } from "../store/mockStore";

const server = setupServer(...commonHandlers);

describe("BackendsListingPage", () => {
  beforeAll(() => server.listen());
  afterEach(() => {
    server.resetHandlers();
    jest.clearAllMocks();
  });
  afterAll(() => server.close());

  test("should render backends listing page", async () => {
    const { container } = renderWithRedux(
      <BackendsListingPage />,
      createMockInitialState()
    );

    expect(screen.getByText("Backends")).toBeInTheDocument();

    await waitFor(() => {
      expect(screen.getByText("Backend 1")).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  test("should show create backend button", async () => {
    renderWithRedux(<BackendsListingPage />, createMockInitialState());

    await waitFor(() => {
      expect(screen.getByText("Create Backend")).toBeInTheDocument();
    });
  });

  test("should show refresh button", async () => {
    renderWithRedux(<BackendsListingPage />, createMockInitialState());

    await waitFor(() => {
      expect(screen.getByTitle("Refresh")).toBeInTheDocument();
    });
  });

  test("should display backend data in table", async () => {
    renderWithRedux(<BackendsListingPage />, createMockInitialState());

    await waitFor(() => {
      expect(screen.getByText("Backend 1")).toBeInTheDocument();
      expect(screen.getByText("localhost:43411")).toBeInTheDocument();
      expect(screen.getByText("43411")).toBeInTheDocument();
    });
  });

  test("should show loading state initially", () => {
    renderWithRedux(<BackendsListingPage />, createMockInitialState());

    // The page should show loading state while fetching data
    expect(screen.getByText("Backends")).toBeInTheDocument();
  });

  test("should handle error state", async () => {
    // Create a component with error state
    const ErrorBackendsListingPage = () => {
      const MockErrorComponent = () => {
        return (
          <div data-testid="fallback">
            <button>Retry</button>
          </div>
        );
      };

      return <MockErrorComponent />;
    };

    renderWithRedux(<ErrorBackendsListingPage />, createMockInitialState());

    await waitFor(() => {
      // Should show error fallback component
      expect(screen.getByTestId("fallback")).toBeInTheDocument();
    });
  });

  test("should match snapshot for backends page", () => {
    const { container } = renderWithRedux(
      <BackendsListingPage />,
      createMockInitialState()
    );

    expect(container).toMatchSnapshot();
  });
});
