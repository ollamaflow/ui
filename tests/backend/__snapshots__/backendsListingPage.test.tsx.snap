// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackendsListingPage should match snapshot for backends page 1`] = `
<div>
  <main
    class="ant-layout-content pageContainer css-dev-only-do-not-override-1vjf2v5"
  >
    <div
      class="pageTitleContainer ant-flex css-dev-only-do-not-override-1vjf2v5 ant-flex-wrap-wrap ant-flex-align-center ant-flex-justify-space-between ant-flex-gap-small"
    >
      <span
        class="ant-typography css-dev-only-do-not-override-1vjf2v5"
        data-testid="heading"
        style="font-weight: 600; font-size: 16px;"
      >
        <div
          class="ant-flex css-dev-only-do-not-override-1vjf2v5 ant-flex-align-center"
          style="gap: 10px;"
        >
          <span
            class="ant-typography css-dev-only-do-not-override-1vjf2v5"
          >
            Backends
          </span>
          <span
            aria-label="reload"
            class="anticon anticon-reload"
            role="img"
            tabindex="-1"
            title="Refresh"
          >
            <svg
              aria-hidden="true"
              data-icon="reload"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M909.1 209.3l-56.4 44.1C775.8 155.1 656.2 92 521.9 92 290 92 102.3 279.5 102 511.5 101.7 743.7 289.8 932 521.9 932c181.3 0 335.8-115 394.6-276.1 1.5-4.2-.7-8.9-4.9-10.3l-56.7-19.5a8 8 0 00-10.1 4.8c-1.8 5-3.8 10-5.9 14.9-17.3 41-42.1 77.8-73.7 109.4A344.77 344.77 0 01655.9 829c-42.3 17.9-87.4 27-133.8 27-46.5 0-91.5-9.1-133.8-27A341.5 341.5 0 01279 755.2a342.16 342.16 0 01-73.7-109.4c-17.9-42.4-27-87.4-27-133.9s9.1-91.5 27-133.9c17.3-41 42.1-77.8 73.7-109.4 31.6-31.6 68.4-56.4 109.3-73.8 42.3-17.9 87.4-27 133.8-27 46.5 0 91.5 9.1 133.8 27a341.5 341.5 0 01109.3 73.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.6 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c-.1-6.6-7.8-10.3-13-6.2z"
              />
            </svg>
          </span>
        </div>
      </span>
      <button
        class="ant-btn css-dev-only-do-not-override-1vjf2v5 ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
        type="button"
      >
        <span
          class="ant-btn-icon"
        >
          <span
            aria-label="plus"
            class="anticon anticon-plus"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="plus"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
              />
              <path
                d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8z"
              />
            </svg>
          </span>
        </span>
        <span>
          Create Backend
        </span>
      </button>
    </div>
    <div
      class="pageContent"
    >
      <div
        class="ant-table-wrapper css-dev-only-do-not-override-1vjf2v5"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-1vjf2v5"
        >
          <div>
            <div
              aria-busy="true"
              aria-live="polite"
              class="ant-spin ant-spin-spinning css-dev-only-do-not-override-1vjf2v5"
            >
              <span
                class="ant-spin-dot-holder"
              >
                <span
                  class="ant-spin-dot ant-spin-dot-spin"
                >
                  <i
                    class="ant-spin-dot-item"
                  />
                  <i
                    class="ant-spin-dot-item"
                  />
                  <i
                    class="ant-spin-dot-item"
                  />
                  <i
                    class="ant-spin-dot-item"
                  />
                </span>
              </span>
            </div>
          </div>
          <div
            class="ant-spin-container ant-spin-blur"
          >
            <div
              class="ant-table ant-table-middle ant-table-empty css-dev-only-do-not-override-1vjf2v5 ant-table-scroll-horizontal"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                  style="overflow-x: auto; overflow-y: hidden;"
                >
                  <table
                    style="width: 1600px; min-width: 100%; table-layout: auto;"
                  >
                    <colgroup>
                      <col
                        style="width: 150px;"
                      />
                      <col
                        style="width: 200px;"
                      />
                      <col
                        style="width: 150px;"
                      />
                      <col
                        style="width: 80px;"
                      />
                      <col
                        style="width: 120px;"
                      />
                      <col
                        style="width: 100px;"
                      />
                      <col
                        style="width: 150px;"
                      />
                      <col
                        style="width: 120px;"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                        >
                          Identifier
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                        >
                          Name
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                        >
                          Hostname:Port
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                          style="text-align: center;"
                        >
                          SSL
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                        >
                          Health Check URL
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                          style="text-align: center;"
                        >
                          Status
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                        >
                          Created
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                          style="text-align: center;"
                        >
                          Actions
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        aria-hidden="true"
                        class="ant-table-measure-row"
                        style="height: 0px; font-size: 0px;"
                      >
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-placeholder"
                      >
                        <td
                          class="ant-table-cell"
                          colspan="8"
                        >
                          <div
                            class="css-dev-only-do-not-override-1vjf2v5 ant-empty ant-empty-normal"
                          >
                            <div
                              class="ant-empty-image"
                            >
                              <svg
                                height="41"
                                viewBox="0 0 64 41"
                                width="64"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <title>
                                  No data
                                </title>
                                <g
                                  fill="none"
                                  fill-rule="evenodd"
                                  transform="translate(0 1)"
                                >
                                  <ellipse
                                    cx="32"
                                    cy="33"
                                    fill="#f5f5f5"
                                    rx="32"
                                    ry="7"
                                  />
                                  <g
                                    fill-rule="nonzero"
                                    stroke="#d9d9d9"
                                  >
                                    <path
                                      d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                    />
                                    <path
                                      d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                      fill="#fafafa"
                                    />
                                  </g>
                                </g>
                              </svg>
                            </div>
                            <div
                              class="ant-empty-description"
                            >
                              No data
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`BackendsListingPage should render backends listing page 1`] = `
<div>
  <main
    class="ant-layout-content pageContainer css-dev-only-do-not-override-1vjf2v5"
  >
    <div
      class="pageTitleContainer ant-flex css-dev-only-do-not-override-1vjf2v5 ant-flex-wrap-wrap ant-flex-align-center ant-flex-justify-space-between ant-flex-gap-small"
    >
      <span
        class="ant-typography css-dev-only-do-not-override-1vjf2v5"
        data-testid="heading"
        style="font-weight: 600; font-size: 16px;"
      >
        <div
          class="ant-flex css-dev-only-do-not-override-1vjf2v5 ant-flex-align-center"
          style="gap: 10px;"
        >
          <span
            class="ant-typography css-dev-only-do-not-override-1vjf2v5"
          >
            Backends
          </span>
          <span
            aria-label="reload"
            class="anticon anticon-reload"
            role="img"
            tabindex="-1"
            title="Refresh"
          >
            <svg
              aria-hidden="true"
              data-icon="reload"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M909.1 209.3l-56.4 44.1C775.8 155.1 656.2 92 521.9 92 290 92 102.3 279.5 102 511.5 101.7 743.7 289.8 932 521.9 932c181.3 0 335.8-115 394.6-276.1 1.5-4.2-.7-8.9-4.9-10.3l-56.7-19.5a8 8 0 00-10.1 4.8c-1.8 5-3.8 10-5.9 14.9-17.3 41-42.1 77.8-73.7 109.4A344.77 344.77 0 01655.9 829c-42.3 17.9-87.4 27-133.8 27-46.5 0-91.5-9.1-133.8-27A341.5 341.5 0 01279 755.2a342.16 342.16 0 01-73.7-109.4c-17.9-42.4-27-87.4-27-133.9s9.1-91.5 27-133.9c17.3-41 42.1-77.8 73.7-109.4 31.6-31.6 68.4-56.4 109.3-73.8 42.3-17.9 87.4-27 133.8-27 46.5 0 91.5 9.1 133.8 27a341.5 341.5 0 01109.3 73.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.6 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c-.1-6.6-7.8-10.3-13-6.2z"
              />
            </svg>
          </span>
        </div>
      </span>
      <button
        class="ant-btn css-dev-only-do-not-override-1vjf2v5 ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
        type="button"
      >
        <span
          class="ant-btn-icon"
        >
          <span
            aria-label="plus"
            class="anticon anticon-plus"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="plus"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
              />
              <path
                d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8z"
              />
            </svg>
          </span>
        </span>
        <span>
          Create Backend
        </span>
      </button>
    </div>
    <div
      class="pageContent"
    >
      <div
        class="ant-table-wrapper css-dev-only-do-not-override-1vjf2v5"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-1vjf2v5"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table ant-table-middle css-dev-only-do-not-override-1vjf2v5 ant-table-scroll-horizontal"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                  style="overflow-x: auto; overflow-y: hidden;"
                >
                  <table
                    style="width: 1600px; min-width: 100%; table-layout: auto;"
                  >
                    <colgroup>
                      <col
                        style="width: 150px;"
                      />
                      <col
                        style="width: 200px;"
                      />
                      <col
                        style="width: 150px;"
                      />
                      <col
                        style="width: 80px;"
                      />
                      <col
                        style="width: 120px;"
                      />
                      <col
                        style="width: 100px;"
                      />
                      <col
                        style="width: 150px;"
                      />
                      <col
                        style="width: 120px;"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                        >
                          Identifier
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                        >
                          Name
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                        >
                          Hostname:Port
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                          style="text-align: center;"
                        >
                          SSL
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                        >
                          Health Check URL
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                          style="text-align: center;"
                        >
                          Status
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                        >
                          Created
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                          style="text-align: center;"
                        >
                          Actions
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        aria-hidden="true"
                        class="ant-table-measure-row"
                        style="height: 0px; font-size: 0px;"
                      >
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="backend-1"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <span
                            aria-describedby="test-id"
                            style="font-family: monospace;"
                          >
                            backend-1
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          Backend 1
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            localhost
                            :
                            43411
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <span
                            class="ant-tag ant-tag-error css-dev-only-do-not-override-1vjf2v5"
                          >
                            No
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            <span
                              class="ant-tag ant-tag-blue mr-0 css-dev-only-do-not-override-1vjf2v5"
                            >
                              GET
                            </span>
                             
                            /health
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <span
                            class="ant-tag ant-tag-success css-dev-only-do-not-override-1vjf2v5"
                          >
                            Active
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          1/1/2021
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <div
                            class="ant-flex css-dev-only-do-not-override-1vjf2v5 ant-flex-align-center ant-flex-justify-space-around"
                          >
                            <a
                              href="/dashboard/backends/edit/backend-1"
                            >
                              <span
                                aria-label="edit"
                                class="anticon anticon-edit"
                                role="img"
                                style="cursor: pointer; color: rgb(24, 144, 255);"
                                title="Edit Backend"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="edit"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                                  />
                                </svg>
                              </span>
                               
                            </a>
                            <span
                              aria-label="delete"
                              class="anticon anticon-delete"
                              role="img"
                              style="cursor: pointer; color: rgb(255, 77, 79);"
                              tabindex="-1"
                              title="Delete Backend"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="delete"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                                />
                              </svg>
                            </span>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
            <ul
              class="ant-pagination ant-pagination-mini ant-table-pagination ant-table-pagination-right css-dev-only-do-not-override-1vjf2v5"
            >
              <li
                class="ant-pagination-total-text"
              >
                1-1 of 1 backends
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-prev ant-pagination-disabled"
                title="Previous Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="left"
                    class="anticon anticon-left"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
              <li
                class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                tabindex="0"
                title="1"
              >
                <a
                  rel="nofollow"
                >
                  1
                </a>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-next ant-pagination-disabled"
                title="Next Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="right"
                    class="anticon anticon-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
              <li
                class="ant-pagination-options"
              >
                <div
                  aria-label="Page Size"
                  class="ant-select ant-select-sm ant-select-outlined ant-pagination-options-size-changer css-dev-only-do-not-override-1vjf2v5 ant-select-single ant-select-show-arrow ant-select-show-search"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-wrap"
                    >
                      <span
                        class="ant-select-selection-search"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-controls="rc_select_TEST_OR_SSR_list"
                          aria-expanded="false"
                          aria-haspopup="listbox"
                          aria-label="Page Size"
                          aria-owns="rc_select_TEST_OR_SSR_list"
                          autocomplete="off"
                          class="ant-select-selection-search-input"
                          id="rc_select_TEST_OR_SSR"
                          role="combobox"
                          type="search"
                          value=""
                        />
                      </span>
                      <span
                        class="ant-select-selection-item"
                        title="10 / page"
                      >
                        10 / page
                      </span>
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
